### PR TITLE
server: make capabilities cache size configurable and add a note when…

### PR DESCRIPTION
**Pull request by jef-n**

### CODE REVIEW

File: src/server/qgscapabilitiescache.cpp [hunk 1]
```diff
@@ -52,12 +53,15 @@ const QDomDocument *QgsCapabilitiesCache::searchCapabilitiesDocument( const QStr
 
 void QgsCapabilitiesCache::insertCapabilitiesDocument( const QString &configFilePath, const QString &key, const QDomDocument *doc )
 {
-  if ( mCachedCapabilities.size() > 40 )
+  static const int cacheSize = qEnvironmentVariable( "QGIS_SERVER_CAPABILITIES_CACHE_SIZE", "40" ).toInt();
```
Comment (elpaso) [inline src/server/qgscapabilitiescache.cpp:56]: You may want to use toInt( &ok ) and check if ok is true, raise a warning if not.

**Review by elpaso — APPROVED** [review]
LGTM, see if my comment makes sense.

**Comment by rldhont** [comment]
@dmarteau can you take a look.

**Review by elpaso — CHANGES_REQUESTED** [review]
Sorry, having a second thought here: you should use QgsServerSettingsEnv:: API for the server variables.

File: python/server/auto_generated/qgsserversettings.sip.in [hunk 2]
```diff
@@ -357,6 +358,15 @@ by setting the environment variable QGIS_SERVER_APPLICATION_NAME.
 Returns the string representation of a setting.
 
 .. versionadded:: 3.16
+%End
+
+    int capabilitiesCacheSize() const;
+%Docstring
+Returns the QGIS Server capabilities cache size.
```
Comment (elpaso) [inline python/server/auto_generated/qgsserversettings.sip.in:365]: Can you specify the uom for the cache size (number of cached capabilities)?

**Review by elpaso — APPROVED** [review]

**Pull request merged**